anndata/_core/aligned_mapping.py

```diff
@@ -47,7 +49,9 @@ def _ipython_key_completions_(self) -> List[str]:
     def _validate_value(self, val: V, key: str) -> V:
         """Raises an error if value is invalid"""
         for i, axis in enumerate(self.axes):
-            if self.parent.shape[axis] != val.shape[i]:
+            if (not isinstance(val, AwkArray)) and (  # don't validate awk arrays
+                self.parent.shape[axis] != dim_len(val, i)
+            ):
```

Comment:
why not validate? I think we should only allow awkward arrays that have a fixed length in the dimensions they are aligned to an AnnData axis.